catalogd/internal/serverutil/serverutil_tests.go

```diff
@@ -0,0 +1,254 @@
+package serverutil
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"sigs.k8s.io/yaml"
+)
+
+// {
+// 	name: "provides gzipped content for the path /catalogs/test-catalog/api/v1/all with size > 1400 bytes",
+// 	setupStore: func() error {
+// 		return writeFile(filepath.Join(store.RootDir, "test-catalog3", "catalog.jsonl"), []byte(testCompressableJSON), 0600)
+// 	},
+// 	expectStatusOK:  true,
+// 	expectedContent: testCompressableJSON,
+// 	URLPath:         "/catalogs/test-catalog3/api/v1/all",
+// },
+
+// jsonLineFormattedCompresableJSON, err := generateJSONLines([]byte(testCompressableJSON))
+// 	require.NoError(t, err)
+// {
+// 	name: "Provides JSON-lines format for the served JSON catalog",
+// 	setupStore: func() error {
+// 		unpackResultFS := &fstest.MapFS{
+// 			"catalog.json": &fstest.MapFile{Data: []byte(testCompressableJSON), Mode: os.ModePerm},
+// 		}
+// 		return store.Store(context.Background(), "test-catalog4", unpackResultFS)
+// 	},
+// 	expectStatusOK:  true,
+// 	expectedContent: jsonLineFormattedCompresableJSON,
+// 	URLPath:         "/catalogs/test-catalog4/api/v1/all",
+// },
+
+// yamlData, err := makeYAMLFromConcatenatedJSON([]byte(testCompressableJSON))
+// require.NoError(t, err)
+// jsonLineFormattedYamlData, err := generateJSONLines(yamlData)
+// require.NoError(t, err)
+// {
+// 	name:            "Provides JSON-lines format for the served YAML catalog",
+// 	expectStatusOK:  true,
+// 	expectedContent: jsonLineFormattedYamlData,
+// 	URLPath:         fmt.Sprintf("%s/test-catalog/api/v1/all", urlPrefix),
+// 	setupStore: func() error {
+// 		yamlData, err := makeYAMLFromConcatenatedJSON([]byte(testCompressableJSON))
+// 		if err != nil {
+// 			return err
+// 		}
+// 		unpackResultFS := &fstest.MapFS{
+// 			"catalog.yaml": &fstest.MapFile{Data: yamlData, Mode: os.ModePerm},
+// 		}
+// 		err = store.Store(context.Background(), "test-catalog", unpackResultFS)
+// 		if err != nil {
+// 			return err
+// 		}
+// 		return err
+// 	},
+// },
+
+// {
+// 	name: "Ignores accept-encoding for the path /catalogs/test-catalog/api/v1/all with size < 1400 bytes",
+// 	setupStore: func() error {
+// 		return writeFile(filepath.Join(store.RootDir, "test-catalog2", "catalog.jsonl"), []byte(`{"foo":"bar"}`), 0600)
```

Comment:
I know existing tests violate this rule, but one thing I think I would fix is to avoid writing to the underlying filesystem, because it is pretty fragile. I think we can probably test as a black box:
1. Check `ContentExists(foo)` is false
1. `Store(foo)`
2. Query various endpoints with positive and negative tests for `foo`, `bar` catalogs, but also just different path prefixes, etc.
3. Check `ContentExists(foo)` is true
4. `Delete(foo)`
5. Query `foo` endpoints and expect 404
6. Check `ContentExists(foo)` is false.

Reply:
+1, I've added this test in `localdir_test.go`, and avoided writing directly to the file system (using `store.Store` instead)